apps/src/code-studio/pd/form_components_func/FormController.jsx

```diff
@@ -183,63 +184,68 @@ const FormController = props => {
    * @return {boolean} true if this page is valid, false if any required fields
    *         are missing
    */
-  const validatePageRequiredFields = pageIndex => {
-    if (pageIndex < 0 || pageIndex >= pageComponents.length) {
-      throw `Invalid page index ${pageIndex}`;
-    }
+  const validatePageRequiredFields = useCallback(
```

Comment:
Can you explain what necessitated the change for `useCallback`? Not sure I understand it ...

Reply:
`validatePageRequiredFields` is a dependency of a `useEffect` call (meaning that if `validatePageRequiredFields` changes, the effect will change), if we define `validatePageRequiredFields` without using `useCallback` it will get re-defined for every render and therefore useEffect would get called more often than it should. `useCallback` will memoize the function so that it's definition doesn't change unless the dependencies change. https://reactjs.org/docs/hooks-reference.html#usecallback Hopefully that explanation made sense. The linter will also tell you when you're in this scenario and should use `useCallback`

Reply:
The docs I linked have a more clear explanation